src/snowflake/cli/plugins/snowpark/commands.py

```diff
@@ -308,30 +313,92 @@ def _deploy_single_object(
+    log.info("Writing %s file", file_path.path)
+    with file_path.open("w", encoding="utf-8") as f:
+        for req in requirements:
+            f.write(f"{req.line}\n")
+
+
 @app.command("build")
 @with_project_definition("snowpark")
 def build(
-    pypi_download: YesNoAsk = PyPiDownloadOption,
-    check_anaconda_for_pypi_deps: bool = CheckAnacondaForPyPiDependencies,
-    package_native_libraries: YesNoAsk = PackageNativeLibrariesOption,
+    ignore_anaconda: bool = IgnoreAnacondaOption,
+    allow_shared_libraries: bool = AllowSharedLibrariesOption,
+    index_url: Optional[str] = IndexUrlOption,
+    skip_version_check: bool = SkipVersionCheckOption,
+    deprecated_package_native_libraries: YesNoAsk = deprecated_allow_native_libraries_option(
+        "--package-native-libraries"
+    ),
+    deprecated_check_anaconda_for_pypi_deps: bool = DeprecatedCheckAnacondaForPyPiDependencies,
+    _deprecated_pypi_download: YesNoAsk = deprecated_pypi_download_option,
     **options,
 ) -> CommandResult:
     """
     Builds the Snowpark project as a `.zip` archive that can be used by `deploy` command.
     The archive is built using only the `src` directory specified in the project file.
     """
+    if not deprecated_check_anaconda_for_pypi_deps:
+        ignore_anaconda = True
+    # TODO: yes/no/ask logic should be removed in 3.0
+    allow_shared_libraries_yesnoask = {
+        True: YesNoAsk.YES,
+        False: YesNoAsk.NO,
+    }[allow_shared_libraries]
+    if deprecated_package_native_libraries != YesNoAsk.NO:
+        allow_shared_libraries_yesnoask = deprecated_package_native_libraries
+
     paths = SnowparkPackagePaths.for_snowpark_project(
         project_root=SecurePath(cli_context.project_root),
         snowpark_project_definition=cli_context.project_definition,
     )
     log.info("Building package using sources from: %s", paths.source.path)
 
-    snowpark_package(
-        paths=paths,
-        pypi_download=pypi_download,  # type: ignore[arg-type]
-        check_anaconda_for_pypi_deps=check_anaconda_for_pypi_deps,
-        package_native_libraries=package_native_libraries,  # type: ignore[arg-type]
-    )
+    with SecurePath.temporary_directory() as packages_dir:
+        if paths.defined_requirements_file.exists():
+            log.info("Resolving any requirements from requirements.txt...")
+            requirements = package_utils.parse_requirements(
+                requirements_file=paths.defined_requirements_file,
+            )
+
+            download_result = package_utils.download_unavailable_packages(
+                requirements=requirements,
+                target_dir=packages_dir,
+                ignore_anaconda=ignore_anaconda,
+                skip_version_check=skip_version_check,
+                pip_index_url=index_url,
+                allow_shared_libraries=allow_shared_libraries_yesnoask,
+            )
+            if not download_result.download_successful:
```

Comment:
```suggestion
            if not download_result.succeeded:
```
WDYT? To avoid download download